api/list.go

```diff
@@ -125,19 +128,35 @@ func (c *Client) ListObjects(ctx context.Context, list runtimeclient.ObjectList,
 		return fmt.Errorf("error when searching for projects: %w", err)
 	}
 
+	projectsSize := len(projects)
+	var wg sync.WaitGroup
+	wg.Add(projectsSize)
+	ch := make(chan reflect.Value, projectsSize)
+
```

Comment:
Maybe limit the number of concurrent go routines as well.

Reply:
I didn't limit it for now. Since we already have the rate limit throttling of the k8s client library it doesn't make much sense. We can easily have 100s if not 1000s of concurrently running go routines without any issues.